web/src/main/java/org/cbioportal/web/AlterationEnrichmentController.java

```diff
@@ -66,11 +66,27 @@
         Select<CNA> cnaEventTypes = allOptionsSelected(alterationEventTypes.getCopyNumberAlterationEventTypes()) ?
             Select.all() : Select.byValues(selectedCnas);
 
+        Select<String> selectedTiers = Select.byValues(
+            alterationEventTypes.getSelectedTiers().entrySet().stream()
+                .filter(e -> e.getValue())
+                .map(e -> e.getKey()));
+        if (alterationEventTypes.getSelectedTiers().keySet().size() > 0
```

Comment:
I have a question here. Consider there are 3 tiers - tier1, tier2, tier3. If alterationEventTypes.selectedTiers is `{"tier1" : true, "tier2": true}` then is this condition true and selectedTiers is said to have all values?

Reply:
That is correct!

Reply:
isn't that a bug then?

Reply:
How do you mean?

Reply:
this condition is will be true for both
`{"tier1" : true, "tier2": true, "tier3" : true}` and `{"tier1" : true, "tier2": true}` even if there is no information about `tier3`
Does this means the default value is true if tier is not present in request object?

Reply:
Ah I understand now. No, it evaluates whether to search for all types based on all keys being true. So this logic expects the frontend to send over a complete map of all options. If the frontend does not do this, there wil be a problem indeed. There is no other way for us to evaluate whether all options are selected I can see. Shall we discuss this with @alisman to see whether this a correct approach?

Reply:
This shouldn't be an issue if the request is coming for frontend. But if the user is trying to query API directly then it might be an issue.